File: leetcode-submissions/1381_Design_a_Stack_With_Increment_Operation.cpp
class CustomStack {
public:
stack<int>st;
int size;
    CustomStack(int maxSize) {
        this->size=maxSize;
    }
    
    void push(int x) {
        if(this->st.size()<this->size){
            this->st.push(x);
        }
    }
    
    int pop() {
        if(this->st.size()>0){
            int top=this->st.top();
            this->st.pop();
            return top;
        }
        return -1;
    }
    
    void increment(int k, int val) {
        if(k>this->st.size()){
            stack<int>temp;
            while(not this->st.empty()){
                temp.push(st.top());
                st.pop();
            }
            while(not temp.empty()){
                st.push(temp.top()+val);
                temp.pop();
            }
        }else{
            
            stack<int>temp;
            while(not this->st.empty()){
                temp.push(st.top());
                st.pop();
            }
            int cnt=0;
           while(cnt<k){
            this->st.push(temp.top()+val);
            temp.pop();
            cnt++;
           }
           while(not temp.empty()){
            this->st.push(temp.top());
            temp.pop();

           }

        }
    }
};

/**
 * Your CustomStack object will be instantiated and called as such:
 * CustomStack* obj = new CustomStack(maxSize);
 * obj->push(x);
 * int param_2 = obj->pop();
 * obj->increment(k,val);
 */

File: leetcode-submissions/1381_Design_a_Stack_With_Increment_Operation.ts
class CustomStack {
    private stack:number[];
    private size:number ;

    constructor(maxSize: number) {
        this.size=maxSize;
        this.stack=[];
    }

    push(x: number): void {
        if(this.stack.length<this.size){
            this.stack.push(x);
        }
    }

    pop(): number {
        if(this.stack.length>0){
            const topEle=this.stack[this.stack.length-1];
            this.stack.pop();
            return topEle;
        }
        return -1;
    }

    increment(k: number, val: number): void {
  let limit=Math.min(k,this.stack.length);
  for(let i=0;i<limit;i++){
    this.stack[i]+=val;
  }

    }
}

/**
 * Your CustomStack object will be instantiated and called as such:
 * var obj = new CustomStack(maxSize)
 * obj.push(x)
 * var param_2 = obj.pop()
 * obj.increment(k,val)
 */